pkg/deployments/tasks/20230217-gnosis-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,326 @@
+    // We now need to go through an epoch for the votes to be locked in
+    await advanceTime(DAY * 8);
+
+    await gaugeController.checkpoint();
+
+    // Gauge weight is equal to the cap, and controller weight for the gauge is greater than the cap.
+    expect(
+      await gaugeController['gauge_relative_weight(address,uint256)'](gauge.address, await currentWeekTimestamp())
+    ).to.be.gt(weightCap);
+    expect(await gauge.getCappedRelativeWeight(await currentTimestamp())).to.equal(weightCap);
+  });
+
+  it('mint & bridge tokens', async () => {
+    // The gauge has votes for this week, and it will mint the first batch of tokens. We store the current gauge
+    // relative weight, as it will change as time goes by due to vote decay.
+    const firstMintWeekTimestamp = await currentWeekTimestamp();
+    //const gaugeRelativeWeight = await gaugeController['gauge_relative_weight(address)'](gauge.address);
+
+    const calldata = gauge.interface.encodeFunctionData('checkpoint');
+
+    // Even though the gauge has relative weight, it cannot mint yet as it needs for the epoch to finish
+    const zeroMintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+    expectEvent.inIndirectReceipt(await zeroMintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp.sub(WEEK), // Process past week, which had zero votes
+      periodEmissions: 0,
+    });
+    // No token transfers are performed if the emissions are zero, but we can't test for a lack of those
+
+    await advanceTime(WEEK);
+
+    // The gauge should now mint and send all minted tokens to the Gnosis bridge
+    const mintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+    const event = expectEvent.inIndirectReceipt(await mintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp,
+    });
+    const actualEmissions = event.args.periodEmissions;
+
+    // The amount of tokens minted should equal the weekly emissions rate times the relative weight of the gauge
+    const weeklyRate = (await BALTokenAdmin.getInflationRate()).mul(WEEK);
+
+    const expectedEmissions = weightCap.mul(weeklyRate).div(FP_ONE);
+    expectEqualWithError(actualEmissions, expectedEmissions, 0.001);
+
+    // Tokens are minted for the gauge
+    expectTransferEvent(
+      await mintTx.wait(),
+      {
+        from: ZERO_ADDRESS,
+        to: gauge.address,
+        value: actualEmissions,
+      },
+      BAL
+    );
+
+    // And the gauge then deposits those in the predicate via the bridge mechanism
+    const bridgeInterface = new ethers.utils.Interface([
+      'event TokensBridgingInitiated(address indexed token, address indexed sender, uint256 value, bytes32 indexed messageId)',
+    ]);
+
+    expectEvent.inIndirectReceipt(await mintTx.wait(), bridgeInterface, 'TokensBridgingInitiated', {
+      token: BAL,
+      sender: gauge.address,
+      value: actualEmissions,
+    });
```

Comment:
Given that it's hard to do local tests for bridges, we're a bit more strict in the procedure for merging fork test PRs that interact with them.

Please manually initiate the bridge process from an EOA, calling the same function our code does with the same arguments (`_gnosisBridge.relayTokens`), and submit the explorer link here, showing that it emits the same event we test for here with the same arguments. Ideally we'd also see e.g. the recipient.

Then, once the bridging is complete, also paste here the explorer link to the gnosis transaction where the tokens landed, so we can verify that this call indeed works.

Reply:
Call to relayTokens, showing the event:
https://etherscan.io/tx/0x6a0dcbf72db757f83bf1c9b42e5f940c31e3240479614635fcbe5a5f72091692

Gnosis chain receipt:
https://gnosisscan.io/tx/0x983bede2006468b5fbc67204068c2b1c9492be65530ce7f34e82841703714294